ocdev/plugins/setup/setup.py

```diff
@@ -167,5 +167,11 @@ def clone_repo(self, url, branch='master', directory=None, no_history=False):
         if directory:
             cmd.append(directory)
 
-
-        return check_call(cmd)
\ No newline at end of file
+        assert not os.path.exists('{0}/core'.format(os.getcwd())), 'cannot clone, {0}/core already exists'.format(os.getcwd())
+        assert os.access(os.getcwd(), os.W_OK), 'Insufficient privilege to write to this directory'
```

Comment:
Ah I see, the current directory is passed in here https://github.com/WillBrennan/ocdev/blob/master/ocdev/plugins/setup/setup.py#L64

The target_dir is where it should be cloned to relative to the current path https://github.com/WillBrennan/ocdev/blob/master/ocdev/plugins/setup/setup.py#L67

I wonder if the best way for all these checks is to just don't do them and find a way to print the git error message nicely and abort if the return code is not success